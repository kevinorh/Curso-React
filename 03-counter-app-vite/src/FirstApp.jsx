import PropTypes from 'prop-types';

export const FirstApp = ( {
  title, 
  subtitle,
  name
}) => {
  // console.log({title});
  return (
    <>
        <h1 data-testid="test-title"> {title} </h1>
        <h2>{subtitle}</h2>
        <h2>{subtitle}</h2>
        <h3>{name}</h3>
    </>
  )
}

FirstApp.propTypes ={
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string
}

FirstApp.defaultProps ={
  // title: 'No hay ningun t√≠tulo',
  subtitle: 'No hay subtitulo',
  name: 'Kevin Ore'
}