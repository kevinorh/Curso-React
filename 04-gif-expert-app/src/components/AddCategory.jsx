import { useState } from "react"
import PropTypes from 'prop-types'

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const onChangeHandler = ({ target }) => {
        // console.log(event.target.value)
        setInputValue(target.value)
    }

    const onSubmit = (event) => {
        // console.log('Hola Mundo desde submit')
        event.preventDefault();

        if( inputValue.trim().length <= 1) return;
        
        // setCategories(categories => [...categories, inputValue])
        onNewCategory(inputValue)
        setInputValue('')
        // console.log('Hola Mundo desde submit fin')
    }



    return (
        // <form onSubmit={(event) => onSubmit(event)}>
        <form onSubmit={onSubmit} aria-label="form">
            <input
                type="text"
                placehoder="buscar gifs"
                value={inputValue}
                // onChange={ (event) => onChangeHandler(event)}
                onChange={onChangeHandler}
            />
        </form>
    )
}


AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}