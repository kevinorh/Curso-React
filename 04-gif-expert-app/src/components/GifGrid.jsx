// import { useEffect, useState } from "react"
// import { getGifs } from "../helpers/getGifs"
import { GifGridItem } from "./GifGridItem"; 
import { useFetchGifs } from "../hooks/useFetchGIfs";
import PropTypes from 'prop-types'

export const GifGrid = ({ category }) => {

    // const [images, setImages] = useState([]);

    // const getImages = async () => {
    //     const newImages = await getGifs(category);
    //     setImages(newImages)
    // }

    // useEffect(() => {
    //     getImages();
    // }, [])
    //TODO LO DE ARRIBA SE REEMPLAZA POR UN CUSTOM HOOK AQUI ABAJO
    const {images, isLoading} = useFetchGifs(category);

    console.log({images, isLoading});


    return (
        <>
            <h3>{category}</h3>
            {
                //isLoading? <h2>Cargando..</h2>: null
                isLoading && <h2>Cargando..</h2>
            }
            
            <div className="card-grid">
                {
                    images.map((image) => (
                        // <li key={id}>
                        //     <p>{title}</p>
                        //     <img src={url} />
                        // </li>
                        <GifGridItem key={image.id}
                        //EXPARCIR ELEMENTOS
                        { ...image }
                        />
                    )
                    )
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}