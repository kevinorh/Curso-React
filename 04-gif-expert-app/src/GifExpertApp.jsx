import { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Dragon Ball'])

    const onAddCategory = (newCategory) => {
        // console.log(newCategory)

        const exitingItem = categories.find(category => {
            return category.toLocaleLowerCase().trim() === newCategory.toLocaleLowerCase().trim();
        });
        // console.log(exitingItem);


        if (categories.includes(newCategory) && exitingItem) return;
        //     // setCategories(categories => categories.concat('Rainbow Six Siegue'));
        //     // setCategories(categories.concat('Rainbow Six Siegue'));
        setCategories([newCategory, ...categories]);
    }

    return (
        <>
            {/* Title */}
            <h1>GifExpertApp</h1>
            {/* Input */}
            <AddCategory
                // setCategories = {setCategories}
                // onNewCategory = { (value) => onAddCategory(value) }
                onNewCategory={onAddCategory}
            />
            {/* Gif List */}
            {categories.map((category) =>
            (
                // <div key={category}>
                //     <h3>{category}</h3>
                //     <li>{category}</li>
                // </div>
                <GifGrid
                    key={category}
                    category={category} />
            )
            )
            }
            {/* <li>123</li> */}
            {/* Gif Item */}
        </>
    )
}