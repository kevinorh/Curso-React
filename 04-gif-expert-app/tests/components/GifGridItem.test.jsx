import { render, screen } from "@testing-library/react";
import { GifGridItem } from "../../src/components";


describe('Pruebas en <GifGridItem/>', () => { 
    const title = 'Dragon Ball';
    const url = 'https://dragon-ball-sullca.com/goku.jpg';

    test('should match snapshoot', () => { 
        const { container } = render(<GifGridItem title={ title } url={ url }/>);
        expect( container ).toMatchSnapshot();
     });

     test('debe mostrar la imagen con el URL y el ALT indicado', () =>{
        render(<GifGridItem title={ title } url={ url }/>);
        // screen.debug();
        // expect( screen.getByRole('img').src).toBe(url);
        // expect( screen.getByRole('img').alt).toBe(title);
        const {src, alt} = screen.getByRole('img');
        expect( src ).toBe(url);
        expect( alt ).toBe(title);
     });

     test('debe de tomar el tÃ­tulo en el componente', () => {
        render(<GifGridItem title={ title } url={ url }/>);
        expect( screen.getByText( title )).toBeTruthy();
     });

 }); 