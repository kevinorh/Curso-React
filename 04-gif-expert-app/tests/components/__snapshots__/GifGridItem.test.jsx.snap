// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> should match snapshoot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball"
      src="https://dragon-ball-sullca.com/goku.jpg"
    />
    <p>
      Dragon Ball
    </p>
  </div>
</div>
`;
